3rdparty/python/lockfiles/user_reqs.txt

```diff
@@ -11,13 +11,15 @@
 #   "generated_with_requirements": [
```

Comment:
Maybe revert this file's changes? Ack that this is the script doing it, but the script seems to be behaving differently than we want. In the meantime to automating the fix, manually workaround.

I plan to investigate tomorrow (didn't get a chance today)

Reply:
I'm loathe to mess with auto-generated files. I'll wait for the fix. Thanks for working on it.

Reply:
In case it's not clear from the other PR, I think this is what you'll want to run when making PEX upgrades: `build-support/bin/generate_all_lockfiles.sh --internal --tool lambdex`. Lambdex is because it uses PEX.

Reply:
Ok, that's pretty titchy but good to know. Ideally a user could `./pants <update lock goal> -p pex` and have what lock files need updating figured out. Otherwise it seems its much better to all or nothing rather than pick and choose and choose wrong.

Reply:
I will say, I'll definitely forget this and be relying upon reviewers to catch the forget; as such I think I'll lean towards `--all`, then see the world changed, then remember --internal, then forget --tool then ... ? I'm not sure this system is all that workable in practice to be honest.

Reply:
Given how frequently we update Pex...would you like a `--pex` flag that is shorthand for `--internal --tool lambdex`? I agree this is titchy.

> Ideally a user could ./pants <update lock goal> -p pex and have what lock files need updating figured out

For a user, that is what they do. When you update a tool lockfile, you can either do `./pants generate-lockfiles` (all) or `./pants generate-lockfiles --resolve=black`. When you update a user requirement, you can either do `./pants generate-lockfiles` or `./pants generate-lockfiles --resolve=my_user_resolve`.

It's a weird situation for pantsbuild/pants because we don't just have our internally used lockfiles; we also need to generate default tool lockfiles. Those default tool lockfiles might not be == the tool lockfile we use internally. Hence the script.

Reply:
Also, to be clear: the medium term goal (...maybe short? depends how much of a pain this is) is to operate more like other lockfile producing tools and update lockfiles as a secondary side-effect of running the too: https://github.com/pantsbuild/pants/issues/12014.

That change invalidates a lot of UX questions, so it's worth keeping it in mind if we think the usability of this (even as a temporary strategy) isn't up to snuff.